src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/role/entities/RoleAssignmentForSearch.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.auth.role.entities;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.springframework.util.CollectionUtils.isEmpty;
+import static uk.gov.hmcts.reform.wataskmanagementapi.auth.role.entities.RoleAttributeDefinition.BASE_LOCATION;
+import static uk.gov.hmcts.reform.wataskmanagementapi.auth.role.entities.RoleAttributeDefinition.CASE_ID;
+import static uk.gov.hmcts.reform.wataskmanagementapi.auth.role.entities.RoleAttributeDefinition.CASE_TYPE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.auth.role.entities.RoleAttributeDefinition.JURISDICTION;
+import static uk.gov.hmcts.reform.wataskmanagementapi.auth.role.entities.RoleAttributeDefinition.PRIMARY_LOCATION;
+import static uk.gov.hmcts.reform.wataskmanagementapi.auth.role.entities.RoleAttributeDefinition.REGION;
+
+@Getter
+@EqualsAndHashCode
+@Slf4j
+public class RoleAssignmentForSearch {
+
+    @EqualsAndHashCode.Exclude
+    private String id; //If we keep the original roleAssignment then won't need this.
+    private String roleName;
+    private String classification;
+    private String grantType;
+    private String roleType;
+
+    //Attributes
+    private String jurisdiction;
+    @EqualsAndHashCode.Exclude
+    private Set<String> caseIds = new HashSet<>(); //Case ids is a Set since representative caseIds will have multiple.
+    private String caseType;
+    private String region;
+    private String location;
+    private String primaryLocation;
```

Comment:
Is primaryLocation used for access control?  If it is not then we can group RAs that have different ones.

Reply:
I think it's only baseLocation we care about.